test/plugin/test_out_google_cloud_grpc.rb

```diff
@@ -0,0 +1,955 @@
+    # Using out-of-date config method.
+    setup_gce_metadata_stubs
+    exception_count = 0
+    begin
+      create_grpc_driver(USE_GRPC_CONFIG + PRIVATE_KEY_CONFIG)
+    rescue Fluent::ConfigError => error
+      assert error.message.include? 'Please remove configuration parameters'
+      exception_count += 1
+    end
+    assert_equal 1, exception_count
+  end
+
+  def test_configure_custom_metadata_grpc
+    setup_no_metadata_service_stubs
+    d = create_grpc_driver(USE_GRPC_CONFIG + CUSTOM_METADATA_CONFIG)
+    assert_equal CUSTOM_PROJECT_ID, d.instance.project_id
+    assert_equal CUSTOM_ZONE, d.instance.zone
+    assert_equal CUSTOM_VM_ID, d.instance.vm_id
+  end
+
+  def test_configure_invalid_metadata_missing_project_id_grpc
+    setup_no_metadata_service_stubs
+    exception_count = 0
+    Fluent::GoogleCloudOutput::CredentialsInfo.stubs(:project_id).returns(nil)
+    begin
+      create_grpc_driver(USE_GRPC_CONFIG + CONFIG_MISSING_METADATA_PROJECT_ID)
+    rescue Fluent::ConfigError => error
+      assert error.message.include? 'Unable to obtain metadata parameters:'
+      assert error.message.include? 'project_id'
+      exception_count += 1
+    end
+    assert_equal 1, exception_count
+  end
+
+  def test_configure_invalid_metadata_missing_zone_no_metadata_service_grpc
+    setup_no_metadata_service_stubs
+    exception_count = 0
+    Fluent::GoogleCloudOutput::CredentialsInfo.stubs(:project_id).returns(nil)
+    begin
+      create_grpc_driver(USE_GRPC_CONFIG + CONFIG_MISSING_METADATA_ZONE)
+    rescue Fluent::ConfigError => error
+      assert error.message.include? 'Unable to obtain metadata parameters:'
+      assert error.message.include? 'zone'
+      exception_count += 1
+    end
+    assert_equal 1, exception_count
+  end
+
+  def test_configure_invalid_metadata_missing_vm_id_no_metadata_service_grpc
+    setup_no_metadata_service_stubs
+    exception_count = 0
+    Fluent::GoogleCloudOutput::CredentialsInfo.stubs(:project_id).returns(nil)
+    begin
+      create_grpc_driver(USE_GRPC_CONFIG + CONFIG_MISSING_METADATA_VM_ID)
+    rescue Fluent::ConfigError => error
+      assert error.message.include? 'Unable to obtain metadata parameters:'
+      assert error.message.include? 'vm_id'
+      exception_count += 1
+    end
+    assert_equal 1, exception_count
+  end
+
+  def test_configure_invalid_metadata_missing_all_no_metadata_service_grpc
+    setup_no_metadata_service_stubs
+    Fluent::GoogleCloudOutput::CredentialsInfo.stubs(:project_id).returns(nil)
+    exception_count = 0
+    begin
+      create_grpc_driver(USE_GRPC_CONFIG + CONFIG_MISSING_METADATA_ALL)
+    rescue Fluent::ConfigError => error
+      assert error.message.include? 'Unable to obtain metadata parameters:'
+      assert error.message.include? 'project_id'
+      assert error.message.include? 'zone'
+      assert error.message.include? 'vm_id'
+      exception_count += 1
+    end
+    assert_equal 1, exception_count
+  end
```

Comment:
The above 4 tests seem to be structured in exactly the same way. Might make sense to have one generic test method which each individual test invokes?